src/main/webapp/js/instructorFeedbackResults.js

```diff
@@ -209,15 +209,43 @@ function toggleCollapse(e, pans) {
     var expand = 'Expand';
     var collapse = 'Collapse';
     var panels = pans || $('div.panel-collapse');
+    var shouldExpandPanels = $(e).html().trim().startsWith(expand);
+
+    if (shouldExpandPanels) {
+        // Expand each panel if not yet expanded.
+        // When a panel has class ajax_auto or ajax-response-auto, the panel data has not been loaded yet.
+        // We will load the data by Ajax, and ajax_auto or ajax-response-auto class will be removed.
+        // {@link #toggleCollapse} binds with clicking on both collapse panels button(button for all panels)
+        // and collapse panels button section (button for one panel). Class ajax_auto or ajax-response-auto
+        // is found in {@code panels} only if {@link #toggleCollapse} is invoked by clicking on collapse
+        // panels button.
+        var mightNeedAjaxLoading = $(e).is($('#collapse-panels-button'));
```

Comment:
yup! The panel may or may not need ajax loading. If the panel was clicked before by the instructor, the ajax will load that panel and data for that panel is loaded. If the panel was not clicked before, then we need to load it by ajax first. Thus, for each of the panel, it may or may not need Ajax loading.(The checking here is  to show that if {@code #toggleCollapse} is not called by clicking on collapse panels button, then panel must not need Ajax so we do not need to check for ajax loading for that case )